### ISSUE TITLE
"401 - Request is missing required authentication credential" Error

### ISSUE THREAD

**Issue by kemalozvarol2** [yaza-putu/laravel-google-drive-storage#17]
Following [this tutorial](https://www.youtube.com/watch?v=sKqMHFHZuKU&t=738s&pp=ygUWeWF6YSBwdXR1IGdvb2dsZSBkcml2ZQ%3D%3D) I installed this package and it's been working pretty well for the first couple days. Then I had to take some time off from the project (~1 week) and now when I'm trying to upload or fetch a file from google drive I'm getting this error. 

I have followed the tutorial exactly, put all my credentials (client id, client secret, refresh token, folder id) in my `.env` file. Any idea why am I getting this and how I can fix it?

```
{
  "error": {
    "code": 401,
    "message": "Request is missing required authentication credential. Expected OAuth 2 access token, login cookie or other valid authentication credential. See https://developers.google.com/identity/sign-in/web/devconsole-project.",
    "errors": [
      {
        "message": "Login Required.",
        "domain": "global",
        "reason": "required",
        "location": "Authorization",
        "locationType": "header"
      }
    ],
    "status": "UNAUTHENTICATED",
    "details": [
      {
        "@type": "type.googleapis.com/google.rpc.ErrorInfo",
        "reason": "CREDENTIALS_MISSING",
        "domain": "googleapis.com",
        "metadata": {
          "method": "google.apps.drive.v3.DriveFiles.Create",
          "service": "drive.googleapis.com"
        }
      }
    ]
  }
}
```

This is the code I'm using to request the image file : 

```
public function display(Pic $pic)
    {
        auth()->id() !== $pic->user_id ? abort(404) : null;
 
        $image = Gdrive::get($pic->uuid.$pic->extension);
 
        return response($image->file, 200, ['Content-Type' => 'image/webp']);
    }
```

Edit : I also checked the developer console page again, my `.env` credentials are exactly the same as the one shows in the Credentials section of the project page. Except refresh token which that page doesn't show, but I'm certain that it was correct when I first set up the app. I was able to use it without needing to re-login for about 5 days I worked on this app.

**Comment by kemalozvarol2** [comment]
Update : After opening this issue, I created a new reset token following [the guide](https://github.com/ivanvermeyen/laravel-google-drive-demo/blob/master/README/2-getting-your-refresh-token.md). 

I wasn't sure if I checked the "Auto-refresh the token before it expires" checkbox when I first made the app, this time I made sure I checked it. It gave me a new token which then I put into my app and it worked fine again.

But today, exactly a week later, refresh token must have expired. Because I'm getting the same error as the one in the issue again.

Is there a way to prevent the refresh token from expiring? Because apparently checking that box doesn't prevent it.
